### ISSUE TITLE
Inpaint-and-outpaint issues, The parameters overwrite_width & overwrite_height are not effective

### ISSUE THREAD

**Issue by illusiona** [mrhan1993/Fooocus-API#231]
参数如下
qurey_para
params = {
            "prompt": "	A living room in a light luxury style",
            "negative_prompt": "no people, white edges, white spots, defects, (low quality: 2), white background, "
                               "(low res: 2), no unnecessary things around the product, floor-standing bathroom "
                               "cabinet, (embed: unaestheticXLv31: 0.8), low quality, watermark",
            "style_selections": ['Fooocus V2',
                                 'Fooocus Masterpiece',
                                 'Fooocus Enhance',
                                 'Fooocus Photograph',
                                 'Fooocus Sharp'],
            "performance_selection": "Quality",
            "aspect_ratios_selection": aspect_ratios_selection,
            "image_number": image_number,
            "image_seed": -1,
            "sharpness": 5,
            "guidance_scale": 7,
            "base_model_name": "Actual_Texture_ZY_SDXL.safetensors",
            "refiner_model_name": "",
            "refiner_switch": 0.5,
            "loras": [
                {
                    "model_name": "qinshe.safetensors",
                    "weight": 0.8
                }
            ],
            "advanced_params": {
                "adaptive_cfg": 7,
                "adm_scaler_end": 0.3,
                "adm_scaler_negative": 0.8,
                "adm_scaler_positive": 1.5,
                "canny_high_threshold": 128,
                "canny_low_threshold": 64,
                "controlnet_softness": 0.25,
                "debugging_cn_preprocessor": False,
                "debugging_inpaint_preprocessor": False,
                "disable_preview": False,
                "freeu_b1": 1.01,
                "freeu_b2": 1.02,
                "freeu_enabled": False,
                "freeu_s1": 0.99,
                "freeu_s2": 0.95,
                "inpaint_disable_initial_latent": True,
                "inpaint_engine": "v2.6",
                "inpaint_erode_or_dilate": 0,
                "inpaint_respective_field": 1,
                "inpaint_strength": 1,
                "invert_mask_checkbox": False,
                "mixing_image_prompt_and_inpaint": True,
                "mixing_image_prompt_and_vary_upscale": False,
                "overwrite_width": overwrite_width,
                "overwrite_height": overwrite_high,
                "overwrite_step": 40,
                "overwrite_switch": -1,
                "overwrite_upscale_strength": -1,
                "overwrite_vary_strength": -1,
                "refiner_swap_method": "joint",
                "sampler_name": "dpmpp_3m_sde_gpu",
                "scheduler_name": "karras",
                "skipping_cn_preprocessor": False
            },
            "require_base64": True,
            "async_process": False,
            "webhook_url": "string",
            "input_image": image_base64,
            "input_mask": mask_base64,
            "inpaint_additional_prompt": "",
            "outpaint_selections": [],
            "outpaint_distance_left": -1,
            "outpaint_distance_right": -1,
            "outpaint_distance_top": -1,
            "outpaint_distance_bottom": -1,
            "image_prompts": [{"cn_img": str(TempTools.convert_to_base64(r"C:\Users\Administrator\Downloads\download1.png")).split("'")[1],
                               "cn_stop": 1,
                               "cn_weight": 1,
                               "cn_type": "PyraCanny", },

                              {"cn_img": str(TempTools.convert_to_base64(r"C:\Users\Administrator\Downloads\123.png")).split("'")[1],
                               "cn_stop": 0.5,
                               "cn_weight": 0.4,
                               "cn_type": "ImagePrompt"}
                              ]

        }

---------------------------------------------------------------------------------
All other parameters are working properly, only these two are not enough

**Comment by mrhan1993** [comment]
Ok，I will confirm later

**Comment by illusiona** [comment]
> Ok，I will confirm later

Could you tell me if there is an issue with my parameters?

**Comment by mrhan1993** [comment]
I have check the code, is normal, inpaint mode will resize the result width height to the input image size

**Comment by illusiona** [comment]
> I have check the code, is normal, inpaint mode will resize the result width height to the input image size

This is my fooocus log,I have set the overwrite height and width to 1500, but I cannot find this value in the log 
There is another strange phenomenon, my pictures are all 1000 * 674. But it will automatically expand to 832*1216
Can you help me find the reason in the log below?

---------------------------------------------------------------
[INFO] Mixing image prompt and inpaint is set to True
[Task Queue] Task queue is free, start task, job_id=79fb31c3-2c5b-420c-9f72-56860f8e610f
[Parameters] Adaptive CFG = 7.0
[Parameters] Sharpness = 5.0
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 7.0
[Parameters] Seed = 850541230965133579
[Fooocus] Downloading upscale models ...
[Fooocus] Downloading inpainter ...
[Inpaint] Current inpaint model is E:\Fooocus_win64\Fooocus\models\inpaint\inpaint_v26.fooocus.patch
[Fooocus] Downloading control models ...
[Fooocus] Loading control models ...
extra clip vision: ['vision_model.embeddings.position_ids']
[Parameters] Sampler = dpmpp_3m_sde_gpu - karras
[Parameters] Steps = 40 - 30
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Synthetic Refiner Activated
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Base model loaded: E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors
Synthetic Refiner Activated
Request to load LoRAs [('qinshe.safetensors', 0.8), ('E:\\Fooocus_win64\\Fooocus\\models\\inpaint\\inpaint_v26.fooocus.patch', 1.0)] for model [E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors].
Loaded LoRA [E:\Fooocus_win64\Fooocus\models\loras\qinshe.safetensors] for UNet [E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors] with 722 keys at weight 0.8.
Loaded LoRA [E:\Fooocus_win64\Fooocus\models\inpaint\inpaint_v26.fooocus.patch] for UNet [E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors] with 960 keys at weight 1.0.
Request to load LoRAs [('qinshe.safetensors', 0.8)] for model [E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors].
Loaded LoRA [E:\Fooocus_win64\Fooocus\models\loras\qinshe.safetensors] for UNet [E:\Fooocus_win64\Fooocus\models\checkpoints\Actual_Texture_ZY_SDXL.safetensors] with 722 keys at weight 0.8.
Requested to load SDXLClipModel
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.31 seconds
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] 	A living room in a light luxury style, intricate, elegant, sharp focus, beautiful, highly detailed, cinematic, new bright, colorful, positive, cute, cheerful, rational, creative, color, clear, logical, pleasant, peaceful, relaxed, pure, scientific, solid, thought, ideal, perfect, pretty, innocent, artistic, rich, magnificent, shining
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] 	A living room in a light luxury style, designed tender gorgeous elegant perfect intricate detailed shining divine holy great futuristic enchanted, pretty full innocent massive glowing colossal magical imposing complex highly exposed magic lush open magnificent, bright noble extremely deep vibrant excellent directed cinematic, ambient, dynamic dramatic shiny amazing composition, charming romantic atmosphere, exalted, epic, sublime, sharp focus, awesome, very inspirational
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Fooocus] Image processing ...
Upscaling image with shape (674, 1000, 3) ...
[Fooocus] VAE Inpaint encoding ...
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus] VAE encoding ...
Final resolution is (674, 1000), latent is (832, 1216).
Requested to load CLIPVisionModelWithProjection
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.18 seconds

**Comment by mrhan1993** [comment]
Arbitrary resolution is not supported. Fooocus has a list of supported resolutions, which you see it in this file: [parameters.py](https://github.com/mrhan1993/Fooocus-API/blob/main/fooocusapi/parameters.py)

**Comment by illusiona** [comment]
> Arbitrary resolution is not supported. Fooocus has a list of supported resolutions, which you see it in this file: [parameters.py](https://github.com/mrhan1993/Fooocus-API/blob/main/fooocusapi/parameters.py)

I tried set the "overwrite_width": 1728, "overwrite_height": 576,  "aspect_ratios_selection": "1728*576".  But it seems to have no effect. In fooocus, the output always shows that my final resolution is (674, 1000), latent is (832, 1216).

**Comment by mrhan1993** [comment]
I'll check the processing logic in Fooocus later.

**Comment by illusiona** [comment]
> I'll check the processing logic in Fooocus later.

I am debugging the image resolution change function and found that it has modified the resolution (832.1216) in the Fooocus file, rather than in the API layer.
I saw that the image.py file changed the resolution of the image. And an external function called the size function within it

**Comment by a88852asdw** [comment]
"webhook_url": "string"  How to use
